tests/03-auth.spec.ts

```diff
@@ -0,0 +1,131 @@
+import { chai } from 'mochainon';
+import { ResinAuth } from '../lib/auth';
+import { TokenType } from '../lib/token';
+import apiKeyFixtures from './fixtures/api-keys';
+import jwtFixtures from './fixtures/jwts';
+
+const { expect } = chai;
+const IS_BROWSER = typeof window !== 'undefined';
+
+let dataDirectory;
+if (!IS_BROWSER) {
+	// tslint:disable-next-line no-var-requires
+	const settings = require('resin-settings-client');
+	dataDirectory = settings.get('dataDirectory');
+}
+
+const auth = new ResinAuth({ dataDirectory, tokenKey: 'token-test' });
+
+describe('ResinAuth', () => {
+	beforeEach(() =>
+		// Ensure a clean state before starting
+		auth.removeKey()
+	);
+
+	describe('.hasKey()/.removeKey()', () => {
+		it('should return false when no key was stored', () => {
+			expect(auth.hasKey()).to.eventually.equal(false);
+		});
+
+		it('should return true when any key was successfully stored', () => {
+			auth.setKey(jwtFixtures.empty).then(() => {
```

Comment:
Do we need a return here?

Reply:
Nice catch man, I mean the tests pass in both cases and with the same tests run. But it should be needed, updated